src/Esfa.Vacancy.Register.Api/Orchestrators/SearchApprenticeshipVacanciesOrchestrator.cs

```diff
@@ -21,7 +24,12 @@ public SearchApprenticeshipVacanciesOrchestrator(IMediator mediator, IMapper map
         public async Task<SearchResponse<ApprenticeshipSummary>> SearchApprenticeship(
             SearchApprenticeshipParameters apprenticeSearchParameters)
         {
-            if (apprenticeSearchParameters == null) throw new ValidationException("At least one search parameter is required.");
+            if (apprenticeSearchParameters == null) throw new ValidationException(new List<ValidationFailure>
+            {
+                new ValidationFailure("apprenticeSearchParameters", ErrorMessages.SearchApprenticeships.SearchApprenticeshipParametersIsNull)
+                { ErrorCode = ErrorCodes.SearchApprenticeships.SearchApprenticeshipParametersIsNull}
+            });
+
```

Comment:
For the null check is it possible to create a FluentValidator for SearchApprenticeshipParameters and override the Validate method to do a null check.  Like this? https://stackoverflow.com/a/17095870.  If not then maybe move the validation to its own method.

Reply:
I like the idea of overriding validate, altho.. the point of the null check there is to avoid problems in the mapper immediately after.

So would you be happy with 

```
        public async Task<SearchResponse<ApprenticeshipSummary>> SearchApprenticeship(
            SearchApprenticeshipParameters apprenticeSearchParameters)
        {
            if (apprenticeSearchParameters == null) ThrowValidationException();

            var request = _mapper.Map<SearchApprenticeshipVacanciesRequest>(apprenticeSearchParameters);
            var response = await _mediator.Send(request);
            var results = _mapper.Map<SearchResponse<ApprenticeshipSummary>>(response);
            return results;
        }

        private static void ThrowValidationException()
        {
            throw new ValidationException(new List<ValidationFailure>
            {
                new ValidationFailure("apprenticeSearchParameters", ErrorMessages.SearchApprenticeships.SearchApprenticeshipParametersIsNull)
                    { ErrorCode = ErrorCodes.SearchApprenticeships.SearchApprenticeshipParametersIsNull}
            });
        }
```
It's code bloat caused by the ValidationFailure constructor and the fact that everything else is using validation failures collection in the ValidationException.